### PR TITLE
Resolving #810: ES6 Object Literal Shorthand Detection

### PR DESCRIPTION
**Pull request by timothyeburke**
I had to update some of the decode tests - they no longer made sense once wrapped in the second part of the test and were failing. Let me know if this works for you.

### CODE REVIEW

**Comment by bitwiseman** [comment]
I'm not opposed to test changes in general, but I don't understand why the change to the unencode tests is needed.  I don't think that code path is touched by your change. 

Other than that, it looks reasonable to me. There's also the `myProperty() { ... },` syntax to contend with, but if you want to open a separate issue and solve that later, that'd be fine.

File: js/test/beautify-javascript-tests.js [hunk 1]
```diff
@@ -1277,6 +1277,15 @@ function run_javascript_tests(test_obj, Urlencoded, js_beautify, html_beautify,
             'a.b("c",\n' +
             '    () => d.e\n' +
             ')');
+        
+        // Issue 810 - es6 object literal detection
+        bt(
+            '{\n' +
+            '    a,\n' +
+            '    b: c,\n' +
+            '    d: e,\n' +
```
Comment (bitwiseman) [inline js/test/beautify-javascript-tests.js]: You might as well put both inputs that you listed in the issue report in here.  They were just a little more interesting than this.

File: test/data/javascript.js [hunk 2]
```diff
@@ -1196,6 +1196,17 @@ exports.test_data = {
                     ')'
                 ]
             },
+            {
+                comment: "Issue 810 - es6 object literal detection",
+                unchanged: [
```
Comment (bitwiseman) [inline test/data/javascript.js]: You might as well put both inputs that you listed in the issue report in here. They were just a little more interesting than this - worth covering.

Comment (timothyeburke) [inline test/data/javascript.js]: Will do that.

**Comment by timothyeburke** [comment]
As for the updated test - it's because that test is currently in the format:
```
"foo", 'bar', "baz"
```
and on https://github.com/beautify-web/js-beautify/blob/master/test/template/python-javascript.mustache#L1237 that input gets wrapped in a `{ }` block, meaning it is now
```
{ "foo", 'bar', "baz" }
```
and then beautified and treated as an object literal. Replacing those commas with semicolons changes the test to treat them as single line expressions rather than part of an object literal.

**Comment by bitwiseman** [comment]
Ah, okay.  The javascript version of the test passed? 
If so, then change the content back for both test and in the python test mustache change `bt(...)` to `test_fragment(...)`.  That will avoid the `{ }` wrapping.  

Also, as a last point, an you take your changes and rebase them into one commit?

**Comment by timothyeburke** [comment]
Done and done! My team currently uses this (as part of `grunt-jsbeautifier`) for our es5 files, but I want to get through the issues that are keeping us from using it for our es6. Glad I can help, and thanks for helping me through this one.

**Comment by bitwiseman** [comment]
@timothyeburke - High quality and low friction.  Glad to have you do anymore fixes you choose.  :smile:

**Pull request merged**